docs/MPURegionsUsageGuide.md

```diff
@@ -0,0 +1,299 @@
+			 *  base     : start address for the region
+			 *  size     : size of the region in bytes
+			 *  readonly : true indicates a readonly region
+			 *  execute  : true indicates that the region has execute permission
+			 ****************************************************************************/
+			 
+			void mpu_get_register_value(uint32_t *regs, uint32_t region, uintptr_t base, size_t size, uint8_t readonly, uint8_t execute);
+			```
+			2. In the second step, the values in the array are used to configure the mpu. This second step happens everytime during a context switch.
+			```c
+			/****************************************************************************
+			 * Name: up_set_mpu_app_configuration
+			 *
+			 * Description:
+			 *   Configure MPU for app using register values stored in rtcb's mpu_regs[]
+			 *
+			 * Params:
+			 *  rtcb : Common part of Task Control Block
+			 ****************************************************************************/
+			static inline void up_set_mpu_app_configuration(struct tcb_s *rtcb);
+			```
+3. **Example : Adding new MPU region with RO setting**
+	1. Add your MPU region (say MPU_REG_NUM_XX) in enum MPU_REG_NUM in common mpu.h file.
+	2. To configure this MPU region as RO:
+		- If you require one-time configuration of this MPU region as ReadOnly:
+			- Use any of mpu_xxxx function (ex. mpu_userflash) from architecture specific mpu.h file to configure the region.
+		- Else, if repeated configuration of this MPU region is required:
+			1. Use mpu_get_register_value function to obtain the mpu register values for this region and store them in *regs[]* array (passed as first argument)
+				- For this, you need to declare *(uint32_t)regs[]* array variable for storing mpu register values.
+				- Pass *readonly* argument as *true*, *region* as MPU_REG_NUM_XX and pass appropriate values for base and execute arguments.
+			2. Use these values stored in the array at appropriate stage to configure the mpu.
+				- In case of apps, up_set_mpu_app_configuration function is being used to configure MPU region for app during context switch.
+
+## MPU Usages of platform
+- In Protected Build (CONFIG_BUILD_PROTECTED=y), kernel runs in privileged mode, has access to all memory ranges and defines memory access permissions for unprivileged user code using MPU regions.
+- In TizenRT, below 2 cases arise regarding user code for Protected build:
+### 1. CONFIG_APP_BINARY_SEPARATION=n
+- This means there is a single user binary.
+- In this case, Access and execute permissions are set as below for MPU regions:
+```
+	--------------------------------------------------------------------
+	MPU Region		Memory Segments		Permissions
+	--------------------------------------------------------------------
+	MPU_REG_NUM_UFLASH	TEXT			ROX
+	MPU_REG_NUM_URAM	DATA/HEAP/STACK		RWNX
+	--------------------------------------------------------------------
+```
+
+### 2. CONFIG_APP_BINARY_SEPARATION=y
```

Comment:
This could be `CONFIG_APP_BINARY_SEPARATION enabled (....)`.

Reply:
Updated MPU Usages of platform section with a new format. Please check.